src/app/spaces/space.service.ts

```diff
@@ -61,7 +61,7 @@ export class SpaceService {
           this.buildSpaceIndexMap();
           return space;
         })
-        .catch(this.handleError);
+        .catch((error) => {return this.handleError(error)});
```

Comment:
@VineetReynolds what does this do for you? Arrow functions do not bind its own 'this'. It is the key difference between arrow functions and standard function syntax.